src/components/audio-params.js

```diff
@@ -173,39 +177,40 @@ AFRAME.registerComponent("audio-params", {
 
   update() {
     if (this.audioRef) {
-      this.audioRef.setDistanceModel(this.data.distanceModel);
-      this.audioRef.setRolloffFactor(this.data.rolloffFactor);
-      this.audioRef.setRefDistance(this.data.refDistance);
-      this.audioRef.setMaxDistance(this.data.maxDistance);
-      this.audioRef.panner.coneInnerAngle = this.data.coneInnerAngle;
-      this.audioRef.panner.coneOuterAngle = this.data.coneOuterAngle;
-      this.audioRef.panner.coneOuterGain = this.data.coneOuterGain;
+      if (this.data.audioType === AudioType.PannerNode) {
+        this.audioRef.setDistanceModel(this.data.distanceModel);
+        this.audioRef.setRolloffFactor(this.data.rolloffFactor);
+        this.audioRef.setRefDistance(this.data.refDistance);
+        this.audioRef.setMaxDistance(this.data.maxDistance);
+        this.audioRef.panner.coneInnerAngle = this.data.coneInnerAngle;
+        this.audioRef.panner.coneOuterAngle = this.data.coneOuterAngle;
+        this.audioRef.panner.coneOuterGain = this.data.coneOuterGain;
+      }
       this.data.gain !== undefined && this.updateGain(this.data.gain);
     }
   },
 
   tick() {
     const audio = this.getAudio();
     if (audio) {
-      this.data.position = new THREE.Vector3(
-        audio.panner.positionX.value,
-        audio.panner.positionY.value,
-        audio.panner.positionZ.value
-      );
-      this.data.orientation = new THREE.Vector3(
-        audio.panner.orientationX.value,
-        audio.panner.orientationY.value,
-        audio.panner.orientationZ.value
-      );
-      this.data.distanceModel = audio.panner.distanceModel;
-      this.data.rolloffFactor = audio.panner.rolloffFactor;
-      this.data.refDistance = audio.panner.refDistance;
-      this.data.maxDistance = audio.panner.maxDistance;
-      this.data.coneInnerAngle = audio.panner.coneInnerAngle;
-      this.data.coneOuterAngle = audio.panner.coneOuterAngle;
-      this.data.coneOuterGain = audio.panner.coneOuterGain;
+      if (audio.panner) {
+        this.data.position = new THREE.Vector3(
```

Comment:
The contract for AFRAME components is to use `setAttribute` to set its `data`. If you set values on its `data` directly, subsequent calls to `setAttribute` may overwrite your changes (even if the subsequent call did not specify a value for the same key).

Reply:
Fixed